Improve webdav documentation (#10040)
nextcloud/documentation: developer_manual/client_apis/WebDAV/basic.rst

+-------------------------------+---------------------------------------------------+---------------------------------------------+
| <oc:share-types />            | The list of type share.                           | | ``0`` = User                              |
|                               |                                                   | | ``1`` = Group                             |
|                               |                                                   | | ``3`` = Public link                       |
|                               |                                                   | | ``4`` = Email                             |
|                               |                                                   | | ``6`` = Federated cloud share             |
|                               |                                                   | | ``7`` = Circle                            |
|                               |                                                   | | ``8`` = Guest                             |
|                               |                                                   | | ``9`` = Remote group                      |

> |                               |                                                   | | ``7`` = Circle                            |

Reviewer: Circle is duplicate

|                               |                                                   | | ``10`` = Talk conversation                |
|                               |                                                   | | ``12`` = Deck                             |
|                               |                                                   | | ``15`` = Science mesh                     |
+-------------------------------+---------------------------------------------------+---------------------------------------------+
| <ocs:share-permissions />     | The permissions that the user has over the share. | | ``4`` = Create                            |

> |                               |                                                   | | ``1`` = Read                              |

Reviewer: Sort the permissions by their value, makes it easier to understand how they are used.

|                               |                                                   | | ``2`` = Update                            |
|                               |                                                   | | ``8`` = Delete                            |
|                               |                                                   | | ``16`` = Share                            |
|                               |                                                   | | ``31`` = All                              |
+-------------------------------+---------------------------------------------------+---------------------------------------------+

> | <oc:checksums />              |                                                   |                                             |

Reviewer: What kind of data/format does it require or return?

+-------------------------------+---------------------------------------------------+---------------------------------------------+
| <nc:has-preview />            | Whether a preview of the file is available        | ``true`` or ``false``                       |
+-------------------------------+---------------------------------------------------+---------------------------------------------+
| <oc:size />                   | | Unlike :code:`getcontentlength`,                |                                             |
|                               | | this property also works                        |                                             |
|                               | | for folders, reporting the size                 |                                             |
|                               | | of everything in the folder.                    |                                             |
+-------------------------------+---------------------------------------------------+---------------------------------------------+

> | <nc:rich-workspace />         | This property is provided by the text app.        |                                             |

Reviewer: And what does the property do?

	  </d:prop>
	</d:propfind>

A note about namespaces
^^^^^^^^^^^^^^^^^^^^^^^

When building the body of your DAV request, you will request properties that are available under specific namespace.
It is usual to declare shortcuts for those namespace in the ``d:propfind`` element of the body.

Here is the list of available namespace:

=========================================  ========

>                 Namespace                  Shortcut

Reviewer: Those are officially called namespace URI and namespace prefix, we should use the same names.

And here is how it should look in the body of your DAV request:

.. code-block:: xml

	<?xml version="1.0"?>
		<d:propfind
			xmlns:d="DAV:"
			xmlns:oc="http://owncloud.org/ns"
			xmlns:nc="http://nextcloud.org/ns"
			xmlns:ocs="http://open-collaboration-services.org/ns">
			xmlns:ocm="http://open-cloud-mesh.org/ns">

Supported properties
^^^^^^^^^^^^^^^^^^^^

+-------------------------------+--------------------------------------------------+--------------------------------------------------------------------------+
|           Property            |                   Description                    |                                 Example                                  |
+===============================+==================================================+==========================================================================+
| <d:creationdate />            | The creation date of the node.                   | ``"1970-01-01T00:00:00+00:00"``                                          |
+-------------------------------+--------------------------------------------------+--------------------------------------------------------------------------+

> | <nc:creation_time />          | Same as ``creationdate``, but as a timestamp.    | ``0``                                                                    |

Reviewer: Group all the nc props together

|                               |                                                  | |      locktype: ``write``                                               |
|                               |                                                  | |      lockscope: ``exclusive``                                          |
|                               |                                                  | |      depth: ``0``                                                      |
|                               |                                                  | |      owner: ``Jane Smith``                                             |
|                               |                                                  | |      timeout: ``Infinite``                                             |
|                               |                                                  | |      locktoken: { href: ``urn:uuid:f81de2ad-7f3d-a1b2-4f3c-`` }        |
|                               |                                                  | |      lockroot: { href: ``http://www.example.com/container/`` }         |

> |                               |                                                  | | } ]                                                                    |

Reviewer: This example makes it look like the content would be json, but https://github.com/sabre-io/dav/blob/master/lib/DAV/Xml/Property/LockDiscovery.php writes xml. Could you make the example also use xml?

Author: Update as it is actually a dummy endpoint.

+-------------------------------+--------------------------------------------------+--------------------------------------------------------------------------+

> | <oc:supportedlock />          | | Provide the lock capabilities                  | [ { lockscope: ``exclusive``, locktype: ``write`` } ]                    |

Reviewer: Same as above.

Author: Update as it is actually a dummy endpoint.

| <oc:id />                     | | The fileid namespaced by the instance id.      |                                                                          |
|                               | | Globally unique.                               |                                                                          |
+-------------------------------+--------------------------------------------------+--------------------------------------------------------------------------+
| <oc:fileid />                 | The unique id for the file within the instance.  |                                                                          |
+-------------------------------+--------------------------------------------------+--------------------------------------------------------------------------+
| <oc:downloadURL />            | The URL to download the file.                    |                                                                          |
+-------------------------------+--------------------------------------------------+--------------------------------------------------------------------------+

> | <oc:permissions />            | The permissions that the user has over the file. | | S: Shared                                                              |

Reviewer: Note down that the value is a string that contains all the letters for the given permissions, otherwise someone might thing it's an array or something else.

|                               |                                                  | | ``'group'`` = group folder                                             |
|                               |                                                  | | ``'external'`` = external storage                                      |
|                               |                                                  | | ``'external-session'`` = external storage                              |
+-------------------------------+--------------------------------------------------+--------------------------------------------------------------------------+
| <nc:is-encrypted />           | Whether the folder is end-to-end encrypted.      | | ``0`` for ``false``                                                    |
|                               |                                                  | | ``1`` for ``true``                                                     |
+-------------------------------+--------------------------------------------------+--------------------------------------------------------------------------+

> | <oc:tags />                   | List of user specified tags.                     |                                                                          |

Reviewer: Looking at https://github.com/nextcloud/server/blob/master/apps/dav/lib/Connector/Sabre/TagsPlugin.php and https://github.com/nextcloud/server/blob/master/apps/dav/lib/Connector/Sabre/TagList.php shows that the value is also xml. Can you put down the format in the example?

|                               |                                                  | | ``4`` = Email                                                          |
|                               |                                                  | | ``6`` = Federated cloud share                                          |
|                               |                                                  | | ``7`` = Circle                                                         |
|                               |                                                  | | ``8`` = Guest                                                          |
|                               |                                                  | | ``9`` = Remote group                                                   |
|                               |                                                  | | ``10`` = Talk conversation                                             |
|                               |                                                  | | ``12`` = Deck                                                          |

> |                               |                                                  | | ``15`` = Science mesh                                                  |

Reviewer: More xml as per https://github.com/nextcloud/server/blob/master/apps/dav/lib/Connector/Sabre/ShareTypeList.php, please note down the format.

| <ocs:share-permissions />     | | The permissions that the                       | | ``1`` = Read                                                           |
|                               | | user has over the share.                       | | ``2`` = Update                                                         |
|                               |                                                  | | ``4`` = Create                                                         |
|                               |                                                  | | ``8`` = Delete                                                         |
|                               |                                                  | | ``16`` = Share                                                         |
|                               |                                                  | | ``31`` = All                                                           |
+-------------------------------+--------------------------------------------------+--------------------------------------------------------------------------+

> | <ocm:share-permissions />     | | The permissions that the                       | ``share``, ``read``, ``write``                                           |

Reviewer: The output is a json array:
https://github.com/nextcloud/server/blob/278f6413c9ee6d5c011c59a81d2f8e24a86dfbac/apps/dav/lib/Connector/Sabre/FilesPlugin.php#L322

|                               | | user has over the share.                       |                                                                          |
+-------------------------------+--------------------------------------------------+--------------------------------------------------------------------------+
| <ocm:share-attributes />      | User set attributes.                             | ``[{ "scope" => <string>, "key" => <string>, "enabled" => <bool> }]``    |
+-------------------------------+--------------------------------------------------+--------------------------------------------------------------------------+

> | <nc:sharees />                | The list of share recipient.                     | ``{ id, display-name, share type }``                                    |

Reviewer: https://github.com/nextcloud/server/blob/278f6413c9ee6d5c011c59a81d2f8e24a86dfbac/apps/dav/lib/Connector/Sabre/ShareeList.php uses xml for the value

+-------------------------------+--------------------------------------------------+--------------------------------------------------------------------------+

> | <oc:checksums />              | A list of checksum.                              | ``md5:04c36b75222cd9fd47f2607333029106``                                 |

Reviewer: https://github.com/nextcloud/server/blob/278f6413c9ee6d5c011c59a81d2f8e24a86dfbac/apps/dav/lib/Connector/Sabre/ChecksumList.php uses xml

| <oc:quota-used-bytes />       | Amount of bytes used in the folder.              | ``3950773``                                                              |
+-------------------------------+--------------------------------------------------+--------------------------------------------------------------------------+
| <oc:quota-available-bytes />  | Amount of available bytes in the folder.         | | ``3950773``                                                            |
|                               |                                                  | | ``-1`` Uncomputed free space.                                          |
|                               |                                                  | | ``-2`` Unknown free space.                                             |
|                               |                                                  | | ``-3`` Unlimited free space.                                           |
+-------------------------------+--------------------------------------------------+--------------------------------------------------------------------------+

> | <nc:rich-workspace-file />    | The workspace file.                              |                                                                          |

Reviewer: It's the id of the workspace file: https://github.com/nextcloud/text/blob/5c4bac7e2305be8632b316ca61fc66c2d43805c1/lib/DAV/WorkspacePlugin.php#L125

+-------------------------------+--------------------------------------------------+--------------------------------------------------------------------------+
| <nc:rich-workspace />         | The content of the workspace file.               |                                                                          |
+-------------------------------+--------------------------------------------------+--------------------------------------------------------------------------+

> | <nc:upload_time />            | Date this file was uploaded.                     |                                                                          |

Reviewer: Unix timestamp

		...

Supported properties
^^^^^^^^^^^^^^^^^^^^

+-------------------------------+-------------------------------------------------+--------------------------------------------------------------------------+
|           Property            |                   Description                   |                                 Example                                  |
+===============================+=================================================+==========================================================================+
| <d:creationdate />            | The creation date of the node.                  | ``"1970-01-01T00:00:00+00:00"``                                          |
+-------------------------------+-------------------------------------------------+--------------------------------------------------------------------------+
| <d:getlastmodified />         | The latest modification time.                   | ``"Wed, 20 Jul 2022 05:12:23 GMT"``                                      |
+-------------------------------+-------------------------------------------------+--------------------------------------------------------------------------+

> | <d:getetag />                 | The file's etag.                                | ``"&quot;6436d084d4805&quot;"``                                          |

Reviewer: ```suggestion
| <d:getetag />                 | The file's etag.                                | ``"6436d084d4805"``                                          |
```

Author: On purpose, the etag is wrapped in quotes

Reviewer: Can you tell me why the &qout; is needed?

Author: I have harmonized the use of `"`. It was to denote that it is a string, but not really necessary.

+-------------------------------+-------------------------------------------------+--------------------------------------------------------------------------+
| <d:getcontenttype />          | The mime type of the file.                      | ``"image/jpeg"``                                                         |
+-------------------------------+-------------------------------------------------+--------------------------------------------------------------------------+
| <d:resourcetype />            | Specifies the nature of the resource.           | ``<d:collection />`` for a folder                                        |
+-------------------------------+-------------------------------------------------+--------------------------------------------------------------------------+

> | <d:getcontentlength />        | The size if it is a file in octets.             | ``3030237``                                                              |

Reviewer: I'd say bytes instead of octets, it's way more common

+-------------------------------+-------------------------------------------------+--------------------------------------------------------------------------+
| <oc:permissions />            | | The permissions that the user has over the    | | ``S``: Shared                                                          |
|                               | | file. The value is a string containing        | | ``R``: Shareable                                                       |
|                               | | letters for all available permissions.        | | ``M``: Mounted                                                         |
|                               |                                                 | | ``G``: Readable                                                        |
|                               |                                                 | | ``D``: Deletable                                                       |
|                               |                                                 | | ``NV``: Updateable, Renameable, Moveable                               |

> |                               |                                                 | | ``W``: Updateable (file)                                               |

Reviewer: ```suggestion
|                               |                                                 | | ``W``: Writable (file)                                               |
```

Reviewer: Or is this copied from some other documentation?

Author: I kept the same nomenclature than https://github.com/nextcloud/server/blob/artonge/fix/do_not_allow_to_show_pwd_in_sharing/extra_apps/firstrunwizard/vendor/nextcloud/ocp/OCP/Files/DavUtil.php#L75-L81

Reviewer: I can't see that file :thinking:

Author: Indeed, the link is broken, here is a working one: https://github.com/nextcloud/server/blob/artonge/fix/do_not_allow_to_show_pwd_in_sharing/lib/public/Files/DavUtil.php#L75-L81

| <ocs:share-permissions />     | | The permissions that the                      | | ``1`` = Read                                                           |
|                               | | user has over the share.                      | | ``2`` = Update                                                         |
|                               |                                                 | | ``4`` = Create                                                         |
|                               |                                                 | | ``8`` = Delete                                                         |
|                               |                                                 | | ``16`` = Share                                                         |
|                               |                                                 | | ``31`` = All                                                           |
+-------------------------------+-------------------------------------------------+--------------------------------------------------------------------------+

> | <ocm:share-permissions />     | | The permissions that the user has             | ``['share', 'read', 'write']``                                           |

Reviewer: ```suggestion
| <ocm:share-permissions />     | | The permissions that the user has             | ``["share", "read", "write"]``                                           |
```

Json requires double qoutes

+-------------------------------+-------------------------------------------------+--------------------------------------------------------------------------+
| <oc:checksums />              | An array of checksums.                          | ``<oc:checksum>md5:04c36b75222cd9fd47f2607333029106</oc:checksum>``      |
+-------------------------------+-------------------------------------------------+--------------------------------------------------------------------------+
| <nc:has-preview />            | Whether a preview of the file is available.     | ``true`` or ``false``                                                    |
+-------------------------------+-------------------------------------------------+--------------------------------------------------------------------------+
| <oc:size />                   | | Unlike ``getcontentlength``, this property    | ``127815235``                                                            |
|                               | | also works for folders, reporting the size of |                                                                          |

> |                               | | everything in the folder. Size is in octets.  |                                                                          |

Reviewer: Same as above.

- Example: Create = ``4``; Read = ``1``; Update = ``2``; Delete = ``8``; Share = ``16``; All = ``31``;

checksums
^^^^^^^^^
- Id: ``{http://owncloud.org/ns}checksums``
- Property: ``<oc:checksums />``
- Description:
has-preview
^^^^^^^^^^^
- Id: ``{http://nextcloud.org/ns}has-preview``
- Property: ``<nc:has-preview />``
- Description: Whether a preview of the file is available

size
^^^^
- Id: ``{http://owncloud.org/ns}size``
- Property: ``<oc:size />``
- Description: Unlike :code:`getcontentlength`, this property also works for folders reporting the size of everything in the folder

rich-workspace
^^^^^^^^^^^^^^
- Id: ``{http://nextcloud.org/ns}rich-workspace``
- Property: ``<nc:rich-workspace />``
- Description: This property is provided by the text app

contained-folder-count
^^^^^^^^^^^^^^^^^^^^^^
- Id: ``{http://nextcloud.org/ns}contained-folder-count``
- Property: ``<nc:contained-folder-count />``
- Description: The number of folders directly contained in the folder (not recursively)

contained-file-count
^^^^^^^^^^^^^^^^^^^^
- Id: ``{http://nextcloud.org/ns}contained-file-count``
- Property: ``<nc:contained-file-count />``

> - Description: The number of files directly contained in the folder (not recursively)

Reviewer: I'd prefer a markdown table because the information will look a lot more structured.

Author: I tried a table, but it is unreadable as the container's width is too small. Meaning, we only see the id, and you have to scroll right to read the remaining text.

Reviewer: Ah well :/

Author: Found a solution to split long text in multiple lines. So this is a table now :)

The following properties are supported:

- :code:`{DAV:}getlastmodified`
- :code:`{DAV:}getetag`
- :code:`{DAV:}getcontenttype`
- :code:`{DAV:}resourcetype`
- :code:`{DAV:}getcontentlength`
- :code:`{http://owncloud.org/ns}id` The fileid namespaced by the instance id, globally unique
- :code:`{http://owncloud.org/ns}fileid` The unique id for the file within the instance
- :code:`{http://owncloud.org/ns}favorite`
- :code:`{http://owncloud.org/ns}comments-href`
- :code:`{http://owncloud.org/ns}comments-count`
- :code:`{http://owncloud.org/ns}comments-unread`
- :code:`{http://owncloud.org/ns}owner-id` The user id of the owner of a shared file
- :code:`{http://owncloud.org/ns}owner-display-name` The display name of the owner of a shared file
- :code:`{http://owncloud.org/ns}share-types`
- :code:`{http://owncloud.org/ns}checksums`

> - :code:`{http://nextcloud.org/ns}has-preview`

Reviewer: You need to document the namespaces (because in the table you only use `oc` and `nc`).

Author: I don't think the namespaces are relevant for an app developer. No?

Reviewer: It is, because the prop lookup always depends on the full namespace uri. The name of the namespace is just there for resolving the uri, but can be anything. When you don't set the correct uri the props using it will be ignored. Be aware that I haven't read the WebDAV spec, but this is how it works afaik.

Author: Ah, I get what you mean, you're right this is needed :+1:

+-------------------------------+---------------------------------------------------+---------------------------------------------+
| <oc:owner-display-name />     | The display name of the owner of a shared file.   |                                             |
+-------------------------------+---------------------------------------------------+---------------------------------------------+
| <oc:share-types />            | The list of type share.                           | | ``0`` = User                              |
|                               |                                                   | | ``1`` = Group                             |
|                               |                                                   | | ``3`` = Public link                       |
|                               |                                                   | | ``4`` = Email                             |
|                               |                                                   | | ``6`` = Federated cloud share             |
|                               |                                                   | | ``7`` = Circle                            |

> |                               |                                                   | | ``10`` = Talk conversation                |

Author: Copy-pasted from elsewhere. Will add the other type

+-------------------------------+-------------------------------------------------+--------------------------------------------------------------------------+
| <d:getetag />                 | The file's etag.                                | ``"&quot;6436d084d4805&quot;"``                                          |
+-------------------------------+-------------------------------------------------+--------------------------------------------------------------------------+
| <d:getcontenttype />          | The mime type of the file.                      | ``"image/jpeg"``                                                         |
+-------------------------------+-------------------------------------------------+--------------------------------------------------------------------------+
| <d:resourcetype />            | Specifies the nature of the resource.           | ``<d:collection />`` for a folder                                        |
+-------------------------------+-------------------------------------------------+--------------------------------------------------------------------------+

> | <d:getcontentlength />        | The size if it is a file.                       | ``3030237``                                                              |

Reviewer: Doesn't say what the unit is

|                               | |                                               | | ``</d:lockentry>``                                                     |
+-------------------------------+-------------------------------------------------+--------------------------------------------------------------------------+
| <oc:id />                     | | The fileid namespaced by the instance id.     |                                                                          |
|                               | | Globally unique.                              |                                                                          |
+-------------------------------+-------------------------------------------------+--------------------------------------------------------------------------+
| <oc:fileid />                 | The unique id for the file within the instance. |                                                                          |
+-------------------------------+-------------------------------------------------+--------------------------------------------------------------------------+

> | <oc:downloadURL />            | The URL to download the file.                   |                                                                          |

Reviewer: These could have some example values

Author: No example as not implemented yet by any app. Added a note about that.

|                               | | letters for all available permissions.        | | ``M``: Mounted                                                         |
|                               |                                                 | | ``G``: Readable                                                        |
|                               |                                                 | | ``D``: Deletable                                                       |
|                               |                                                 | | ``NV``: Updateable, Renameable, Moveable                               |
|                               |                                                 | | ``W``: Updateable (file)                                               |
|                               |                                                 | | ``CK``: Creatable                                                      |
+-------------------------------+-------------------------------------------------+--------------------------------------------------------------------------+

> | <nc:creation_time />          | Same as ``creationdate``, but as a timestamp.   | ``0``                                                                    |

Reviewer: The example could be more meaningful

+-------------------------------+-------------------------------------------------+--------------------------------------------------------------------------+
| <oc:comments-count />         | The number of comments.                         |                                                                          |
+-------------------------------+-------------------------------------------------+--------------------------------------------------------------------------+
| <oc:comments-unread />        | The number of unread comments.                  |                                                                          |
+-------------------------------+-------------------------------------------------+--------------------------------------------------------------------------+
| <oc:owner-id />               | The user id of the owner of a shared file.      |                                                                          |
+-------------------------------+-------------------------------------------------+--------------------------------------------------------------------------+

> | <oc:owner-display-name />     | The display name of the owner of a shared file. |                                                                          |

Reviewer: These could have some example values.

|                               |                                                 | |     ``<nc:type>0</nc:type>``                                           |
|                               |                                                 | | ``</nc:sharee>``                                                       |
+-------------------------------+-------------------------------------------------+--------------------------------------------------------------------------+
| <oc:checksums />              | An array of checksums.                          | ``<oc:checksum>md5:04c36b75222cd9fd47f2607333029106</oc:checksum>``      |
+-------------------------------+-------------------------------------------------+--------------------------------------------------------------------------+
| <nc:has-preview />            | Whether a preview of the file is available.     | ``true`` or ``false``                                                    |
+-------------------------------+-------------------------------------------------+--------------------------------------------------------------------------+

> | <oc:size />                   | | Unlike ``getcontentlength``,                  |                                                                          |

Reviewer: Also example value and missing unit

|                               |                                                 | |   ``<nc:acl-mapping-type>group</nc:acl-mapping-type>``                 |
|                               |                                                 | |   ``<nc:acl-mapping-id>admin</nc:acl-mapping-id>``                     |
|                               |                                                 | |   ``<nc:acl-mapping-display-name>admin</nc:acl-mapping-display-name>`` |
|                               |                                                 | |   ``<nc:acl-mask>20</nc:acl-mask>``                                    |
|                               |                                                 | |   ``<nc:acl-permissions>15</nc:acl-permissions>``                      |
|                               |                                                 | | ``</nc:acl>``                                                          |
+-------------------------------+-------------------------------------------------+--------------------------------------------------------------------------+

> | <nc:inherited-acl-list>       | Array of ACL rules from the parents folders     |                                                                          |

Reviewer: Is it the same as the prop above?
Also missing an example.

|                               | | for all available permissions.                 | | ``M``: Mounted                                                         |
|                               |                                                  | | ``G``: Readable                                                        |
|                               |                                                  | | ``D``: Deletable                                                       |
|                               |                                                  | | ``NV``: Updateable, Renameable, Moveable                               |
|                               |                                                  | | ``W``: Updateable (file)                                               |
|                               |                                                  | | ``CK``: Creatable                                                      |
+-------------------------------+--------------------------------------------------+--------------------------------------------------------------------------+

> | <nc:creation_time />          | Same as ``creationdate``, but as a timestamp.    | ``0``                                                                    |

Reviewer: The example value could be more meaningful

When listing favorites, the request will find all favorites in the folder recursively, all favorites for a user can be found by sending the request to :code:`remote.php/dav/files/user`

.. _rfc4918: https://tools.ietf.org/html/rfc4918

Special Headers
---------------

Request Headers
^^^^^^^^^^^^^^^

You can set some special headers that Nextcloud will interpret.

+-----------------+-----------------------------------------------------------------+----------------------------------+
|     Header      |                           Description                           |             Example              |
+=================+=================================================================+==================================+

> | X-OC-MTime      | | Allow to specify a mtime.                                     | ``1675789581``                   |

Reviewer: ```suggestion
| X-OC-MTime      | | Allow to specify a modified time.                                     | ``1675789581``                   |
```

|                 | | The response will contain the header ``X-OC-MTime: accepted`` |                                  |
|                 | | if the mtime was accepted.                                    |                                  |
+-----------------+-----------------------------------------------------------------+----------------------------------+
| X-OC-CTime      | | Allow to specify a creation time.                             | ``1675789581``                   |
|                 | | The response will contain the header ``X-OC-CTime: accepted`` |                                  |
|                 | | if the mtime was accepted.                                    |                                  |
+-----------------+-----------------------------------------------------------------+----------------------------------+

> | OC-Checksum     | Allow to specify a checksum.                                    |                                  |

Reviewer: Does it mean the server will validate that the uploaded content matches this checksum? At first I didn't get this explanation, so it should be improved. Especially what kind of checksum it needs to get.

Author: No idea yet, will update when I have an answer

Author: Updated with the bare minimum

|                 | | The server will return the hash in a header named either:     |                                  |
|                 | | ``X-Hash-MD5``, ``X-Hash-SHA1``, or ``X-Hash-SHA256``.        |                                  |
+-----------------+-----------------------------------------------------------------+----------------------------------+
| OC-Chunked      | Specify that the sent data is part of a chunk upload.           | ``true``                         |
+-----------------+-----------------------------------------------------------------+----------------------------------+
| OC-Total-Length | | Contains the total size of the file during a chunk upload.    | ``4052412``                      |
|                 | | This allow the server to abort faster if the remaining        |                                  |
|                 | | user's quota is not enough.                                   |                                  |
+-----------------+-----------------------------------------------------------------+----------------------------------+

Response Headers
----------------

> Nextcloud will sometime answer with the following headers:

Reviewer: That sounds like it depends on some random thing. Maybe just leave it out, since it's explained in the table anyway?

| X-OC-MTime      | | Allow to specify a modification time.                         | ``1675789581``                           |
|                 | | The response will contain the header ``X-OC-MTime: accepted`` |                                          |
|                 | | if the mtime was accepted.                                    |                                          |
+-----------------+-----------------------------------------------------------------+------------------------------------------+
| X-OC-CTime      | | Allow to specify a creation time.                             | ``1675789581``                           |
|                 | | The response will contain the header ``X-OC-CTime: accepted`` |                                          |
|                 | | if the mtime was accepted.                                    |                                          |
+-----------------+-----------------------------------------------------------------+------------------------------------------+
| OC-Checksum     | | A checksum that will be stored in the DB.                     | ``md5:04c36b75222cd9fd47f2607333029106`` |

> |                 | | The server will not do any sort of  validation.               |                                          |

Reviewer: https://github.com/nextcloud/desktop/blob/613357af70217f56ee697f6be425200a9f680067/doc/architecture.rst?plain=1#L210

Reviewer: Hm ok, then the doc in the desktop client has to be wrong.
